dlr_vieten/explorer/templates/dlr_vieten_explorer_tolerance_factors.html

```diff
@@ -25,17 +26,202 @@
 .backgrid tbody tr:hover { background-color: #f9f9f9; }
 .jh-root { font-family: "symbola"; }
 .jh-type-string { font-style: normal; }
+.vertical-align { display: flex; align-items: center; }
 </style>
 
 <div class="container">
 
+    <ul class="nav nav-pills">
+        <li role="presentation"><a href='#composition'>Calculate A-Site Composition</a></li>
+        <li role="presentation"><a href='#tolerance_factor'>Calculate Tolerance Factor</a></li>
+        <li role="presentation"><a href='#data'>Data Used</a></li>
+    </ul>
+
+    <h3 id="composition">Calculate A-Site Composition for a given tolerance factor</h3>
+    <div class="row vertical-align">
+        <div class="col-md-2">
+            <div class="input-group">
+                <span class="input-group-addon">t</span>
+                <input type="text" class="form-control" placeholder="1.006" id="t">
+            </div>
+        </div>
+        <div class="col-md-2">
+            <div class="input-group">
+                <span class="input-group-addon">CN12/10</span>
+                <input type="text" class="form-control" placeholder="0.8" id="r">
+            </div>
+        </div>
+        <div class="col-md-2">
+            <select id="selectTM" data-placeholder="add TM">
+                <option></option>
+            </select>
+        </div>
+        <div class="col-md-1">
+            <button type="button" class="btn btn-primary" id="go1">Go</button>
+        </div>
+        <div class="col-md-4">
+        </div>
+    </div>
+
+    <div class="row">
+        <div class="col-md-12" id="listTM"></div>
+    </div>
+
+    <h3 id="tolerance_factor">Calculate Tolerance Factor</h3>
     <div class="row">
-        <h3>Ionic Radii</h3>
-        {{ ionic_radii|safe }}
+        <div class="col-md-12">
+            <p>Coming Soon ...</p>
+        </div>
     </div>
 
+    <h3 id="data">Data used for Calculations</h3>
+    <div class="row">
+        <div class="col-md-12">
+            {{ ionic_radii|safe }}
+        </div>
+    </div>
 
 </div>
 
+<script>
+var nrTMs = 0
```

Comment:
Why does `nrTMs` need to be global and then counted manually in l. 105? Shouldn't l. 176 give the correct number of transition metals?

Reply:
I don't really know what the children.length is but it always ended up being 2, no matter how many transition metals there actually were. That's why I used this counting formalism.

Reply:
That was the problem: https://stackoverflow.com/questions/17296385/why-does-children-length-only-return-2
It should be children().length instead of children.length :)